### PR TITLE
Remove all actions with table users_device from backend.

### PR DESCRIPTION
**Pull request by AndreyTepaykin**
Now, the only actions approved with this table, from Users/device/POST and Users/device/delete (which are legal http requests for modification).
And addind/delete rows only allowed from client side using methods Users.Device.subscribe() and Users.Device.unsubscribe().

### CODE REVIEW

File: platform/plugins/Users/classes/Users.php [hunk 1]
```diff
@@ -903,16 +897,6 @@ static function register(
 					$platform = $app['platform'];
 					$appId = Q::ifset($app, 'appId', null);
 					break;
-				case 'device':
```
Comment (EGreg) [inline platform/plugins/Users/classes/Users.php]: Please restore this. Why did you delete it?

File: platform/plugins/Users/classes/Users.php [hunk 2]
```diff
@@ -667,20 +667,14 @@ static function logout()
...
+
 		if ($user) {
 			Q_Utils::sendToNode(array(
 				"Q/method" => "Users/logout",
 				"sessionId" => $sessionId,
 				"userId" => $user->id,
 				"deviceId" => $deviceId
 			));
-
-			// forget the device for this user/session
```
Comment (EGreg) [inline platform/plugins/Users/classes/Users.php]: Please restore this code, why did you delete it?

File: platform/plugins/Users/handlers/Users/oAuth/post.php [hunk 3]
```diff
@@ -16,15 +16,4 @@ function Users_oAuth_post()
 
 	// Set the session id to the access_token
 	Q_Session::id($params['access_token']);
-	
-	// Add a device, if any
-	if ($deviceId = Q::ifset($_REQUEST, 'deviceId', null)) {
```
Comment (EGreg) [inline platform/plugins/Users/handlers/Users/oAuth/post.php]: We need this! Sometimes the "deviceId" is indicated in the request during oAuth, and then you are supposed to add this device. Restore this code.

File: platform/plugins/Users/classes/Users.php [hunk 4]
```diff
@@ -1032,10 +1016,6 @@ static function register(
 				$user->addMobile($mobileNumber, $sms, array(), $p);
 			}
 		}
-		if (!empty($device)) {
```
Comment (EGreg) [inline platform/plugins/Users/classes/Users.php]: Restore this stuff. We need this when device is specified during Users::register

**Review by EGreg — CHANGES_REQUESTED** [review]
Please restore the call to Users_Device::delete() 

and case 'device'

**Comment by AndreyTepaykin** [comment]
The main idea of my changes, is to use one SINGLE method to subscribe and unsubscribe devices.
Adding row to table users_device is not enough to subscribe device. Need also to write some data to client local storage. And notifications will work only if both sides (client and server) have valid data about device. If remove row from users_device table on server, but don't remove from client local storage, Users.Deice.subscribed() method return true (ie device subscribed), because it check subscription only on client (it don't send request to server to check uysers_device table). So if I run Users.Deice.subscribe() method to subscribe device, it will do NOTHING, because it think that device already subscribed. I got this situation many times when debugging notifications. 
So we need:
1. Register devices on client and server synchronously. Because adding row to DB without adding rows to client local storage - not only useless, but even confusing. Exactly this is waht I want to do when try to use Users.Device.subscribe() and Users.Device.unsubscribe() methods only for registering devices.

2. If refuse from prev point, then need to send request to server in method Users.Device.subscribed() to check users_device table. And run Users.Device.subscribed() everytime user subscribed some stream to check whether device vlid subscribed, and if not - subscribe one. Or without Users.Device.subscribed() run Users.Device.subscribe() immediately to subscribe device every time user subscribed to some stream.

**Comment by AndreyTepaykin** [comment]
Reverted commit "Remove all actions with table users_device from backend". Please look.

**Pull request closed without merging**